### PR TITLE
rlm_couchbase: Update to latest libcouchbase3.3.x

### PR DESCRIPTION
**Pull request by jpereira**
* Add support for libcouchbase3.3.x
* Drop use of libfreeradius deprecated functions.

### CODE REVIEW

File: raddb/mods-available/couchbase [hunk 1]
```diff
@@ -78,39 +83,34 @@ couchbase {
...
-		&Connect-Info		= 'connectInfo'
-		&Acct-Session-Time	= 'sessionTime'
-		&Acct-Input-Packets	= 'inputPackets'
-		&Acct-Output-Packets	= 'outputPackets'
-		&Acct-Input-Octets	= 'inputOctets'
-		&Acct-Output-Octets	= 'outputOctets'
-		&Acct-Input-Gigawords	= 'inputGigawords'
-		&Acct-Output-Gigawords	= 'outputGigawords'
-		&Event-Timestamp	= 'lastUpdated'
+		Acct-Session-Id	= 'sessionId'
```
Comment (jpereira) [inline raddb/mods-available/couchbase:86]: revert all the `&` prefix.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -302,6 +312,10 @@ jobs:
         sudo sh -c 'echo "#!/bin/sh"          >  /usr/bin/ucf'
         sudo sh -c 'echo "shift && cp -v \$@" >> /usr/bin/ucf'
         sudo chmod +x /usr/bin/ucf
+
+        wget -4 -O /tmp/couchbase-release-1.0-amd64.deb https://packages.couchbase.com/releases/couchbase-release/couchbase-release-1.0-amd64.deb
```
Comment (jpereira) [inline .github/workflows/ci.yml:316]: just set the needed source.list server.

File: src/modules/rlm_couchbase/configure.ac [hunk 3]
```diff
@@ -96,6 +96,38 @@ if test x$with_[]modname != xno; then
...
+	    )],
+	    [
+	      AC_MSG_RESULT(yes)
+	    ],
+	    [
+	      AC_MSG_RESULT(no)
+	      AC_MSG_FAILURE([libcouchbase 3.3.x library version does not match header version])
+	    ],
+	    [
+	      AC_MSG_RESULT([cross-compiling (assuming yes)])
```
Comment (jpereira) [inline src/modules/rlm_couchbase/configure.ac:126]: drop the `AC_MSG_RESULT([cross-compiling (assuming yes)])` and reuse similar steps like in the src/lib/ldap

File: src/modules/rlm_couchbase/couchbase.c [hunk 4]
```diff
@@ -193,52 +221,102 @@ void couchbase_http_data_callback(lcb_http_request_t request, lcb_t instance, co
...
+	}
+
+	rc = lcb_get_bootstrap_status(cb_inst);
+	if (rc != LCB_SUCCESS) {
+		ERROR("Unable to bootstrap cluster: %s", lcb_strerror_short(rc));
+		goto err2;
+	}
+
+	/* set general method callbacks */
+	// lcb_install_callback(cb_inst, LCB_CALLBACK_STATS, couchbase_stats_callback);
```
Comment (jpereira) [inline src/modules/rlm_couchbase/couchbase.c:314]: double-check with @arr2036

File: src/modules/rlm_couchbase/couchbase.c [hunk 5]
```diff
@@ -114,45 +129,59 @@ void couchbase_get_callback(lcb_t instance, const void *cookie, lcb_error_t erro
...
 	case LCB_SUCCESS:
+	{
+		char const *bytes;
+		size_t nbytes;
+		uint16_t http_status = 0;
+
+		/* debug */
+		lcb_resphttp_http_status(resp, &http_status);
+
+		DEBUG("(http_data_callback) Got HTTP Status: %d", http_status);
```
Comment (jpereira) [inline src/modules/rlm_couchbase/couchbase.c:168]: Rename to `DEBUG2`

File: redhat/freeradius.spec [hunk 6]
```diff
@@ -561,6 +561,7 @@ export RADIUSD_VERSION_RELEASE="%{release}"
         --without-rlm_eap_ikev2 \
         --without-rlm_sql_firebird \
         --without-rlm_sql_db2 \
+        --without-rlm_couchbase \
```
[the remark this answers is not in the document]
Comment (arr2036) [reply redhat/freeradius.spec:564]: We'll need the debian version too

File: src/modules/rlm_couchbase/configure.ac (hunk 3 above)
Comment (arr2036) [reply src/modules/rlm_couchbase/configure.ac:126]: You need to add text to fail i.e. `fail=$fail libcouchbase > 3.3.x`

File: raddb/mods-available/couchbase [hunk 7]
```diff
@@ -52,6 +52,11 @@ couchbase {
 #		detailed_errcodes = "true"
 	}
 
+	#
+	#  user_key:: Couchbase document key for user documents (`unlang` supported).
```
Comment (alandekok) [inline raddb/mods-available/couchbase:56]: `unlang` is the whole language, just say `dynamic expansions are supported`

**Comment by arr2036** [comment]
Monolithic commits, no explanation of changes... lots of conflicts now anyway, so maybe someone manually merged it.

**Pull request closed without merging**